### ISSUE TITLE
memory leak in VideoCapture

### ISSUE THREAD

**Issue by yanchao12122** [opencv/opencv#12534]
- OpenCV => 3.4.2 and 3.4.3
- Operating System / Platform => Windows10 64 Bit
- Compiler => Visual Studio 2015

In the following code:
Press the key of 'q' for every 3 seconds,   the memory will increase >10M each time.

```
#include <iostream>
#include <opencv2/opencv.hpp>
using namespace std;
using namespace cv;
int main(int argc, char **argv)
{
	cv::VideoCapture cap;
	while (true)
	{
		cv::Mat frame;
		cap.open(0);
		while (cap.read(frame))
		{
			cv::imshow("vidoe", frame);
			if ('q' == cv::waitKey(1))
				break;
		}
		cv::destroyAllWindows();
		cap.release();
	}
	return 1;
}

**Comment by yanchao12122** [comment]
I have already fixed the bug.
In the video I/O source file 'cap_msmf.cpp'
Annotate the lines of code from 980 to 986, and rebuild the opencv.

// readCallback = ComPtr<IMFSourceReaderCallback>(new SourceReaderCB());
// HRESULT hr = srAttr->SetUnknown(MF_SOURCE_READER_ASYNC_CALLBACK, (IMFSourceReaderCallback*)readCallback.Get());
// if (FAILED(hr))
// {
// readCallback.Release();
// continue;
// }

**Comment by alalek** [comment]
Try latest code from 3.4 branch. It should be get fixed here: #12393
